src/main/java/de/tum/in/www1/artemis/domain/Organization.java

```diff
@@ -0,0 +1,122 @@
+package de.tum.in.www1.artemis.domain;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import org.hibernate.annotations.Cache;
+import org.hibernate.annotations.CacheConcurrencyStrategy;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+@Entity
+@Table(name = "organization")
+@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class Organization extends DomainObject {
+
+    @NotNull
+    @Size(max = 100)
+    @Column(name = "name", length = 100)
+    private String name;
+
+    @NotNull
+    @Size(max = 50)
+    @Column(name = "shortName", length = 50)
+    private String shortName;
+
+    @Column(name = "url")
+    private String url;
+
+    @Column(name = "description")
+    private String description;
+
+    @Column(name = "logoUrl")
+    private String logoUrl;
+
+    @NotNull
+    @Column(name = "emailPattern")
+    private String emailPattern;
+
+    @ManyToMany
+    @JoinTable(name = "user_organization", joinColumns = { @JoinColumn(name = "organization_id", referencedColumnName = "id") }, inverseJoinColumns = {
```

Comment:
I would be very careful with using @ManyToMany on both sides. Normally it works like this. The entity with @ManyToMany association is responsible for handling this bidirectional association.**So this is the entity that controls if a realtionship exists. Meaning if you remove it from the collection here the relationship will be removed on the database level. Removing the entity from the @MappedBy side has no effect on the databse level**  I have no idea how this works with manyToMany on both sides

In other words https://stackoverflow.com/questions/5460573/hibernate-bidirectional-manytoone-updating-the-not-owning-side-not-working

This is the way it's supposed to work: the owning side (@ManyToMany) is the one which owns the relationship, and which is responsible of managing it in database. It's the responsibility of the developer (i.e. your responsibility) to make sure your object graph is always coherent : if one side of a relationship is modified, then the other side should be modified as well.

If you don't always do this, you must at least make sure to modify the owning side of the relationship, because Hibernate only cares about the owning side..

Reply:
Quite sure I extensively tested it with **manyToMany** on both sides and didn't show inconsistencies. Problem was I wanted to manage the Organizations from both sides (adding users/courses from the Organization side and adding Organizations along with Course/User entities), which is not possible using **mappedBy**...

Anyway I believe that its better to keep sticking to the **mappedBy** solution, so I did some refactoring in e4c6beb to avoid the double-joins.

P.s. The explanation really helped me out, thanks :rocket: